lib/workers/repository/update/branch/index.ts

```diff
@@ -541,7 +549,9 @@ export async function processBranch(
       // compile commit message with body, which maybe needs changelogs
       if (config.commitBody) {
         if (
-          config.fetchReleaseNotes &&
+          // only fetch if fetchReleaseNotes=pr as we would have already
+          // fetched the notes before executing the postUpgradeTasks if fetchReleaseNotes=branch
+          config.fetchReleaseNotes === 'pr' &&
           needsChangelogs(config, ['commitBody'])
```

Comment:
I thought this section would be removed. @viceice wdyt?

Reply:
yes, this if should be removed

Reply:
Yes, this can be removed and makes sense after the doc change you suggested.